File: ningpop/10828.py
# 2022.04.15
# 풀이 시간 14분 26초
# 채점 결과: 정답
# 시간복잡도: O(N)
# 문제 링크: https://www.acmicpc.net/problem/10828

import sys
input = sys.stdin.readline

n = int(input())

stack = []
for _ in range(n):
    order = input()
    
    token = order[:3]
    if token == "pus":
        if order[3] == "h":
            value = int(order.split()[1])
            stack.append(value)
    elif token == "pop":
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif token == "siz":
        if order[3] == "e":
            print(len(stack))
    elif token == "emp":
        if order[3:5] == "ty":
            if stack:
                print(0)
            else:
                print(1)
    elif token == "top":
        if stack:
            print(stack[-1])
        else:
            print(-1)
    else:
        continue

File: keich_park/10828.kt
fun main() {
    val stack = Array(10000) { 0 }
    var pointer = 0

    val n = readLine()?.toInt() ?: return

    repeat(n) {
        val command = readLine()?.split(' ') ?: return

        when (command[0]) {
            "push" -> stack[pointer++] = command[1].toInt()
            "pop" -> {
                if (pointer > 0) println(stack[pointer-- - 1])
                else println(-1)
            }
            "size" -> println(pointer)
            "empty" -> println(if (pointer == 0) 1 else 0)
            "top" -> {
                if (pointer > 0) println(stack[pointer - 1])
                else println(-1)
            }
        }
    }
}